app/lib/actions/katello/capsule_content/sync.rb

```diff
@@ -27,17 +27,31 @@ def plan(capsule_content, options = {})
               need_updates.each do |repo|
                 plan_action(Pulp::Repository::Refresh, repo, capsule_id: capsule_content.capsule.id)
               end
-
               plan_action(ConfigureCapsule, capsule_content, environment, content_view)
+              sync_repos_to_capsule(capsule_content, repository_ids)
+              plan_action(RemoveOrphans, :capsule_id => capsule_content.capsule.id)
```

Comment:
The remove orphans task  is kicked off at the end of a sync. That way when a user changes environments, their content gets cleaned up on the next sync, freeing up disk space.